src/conn/conn_api.c

```diff
@@ -2199,6 +2201,8 @@ wiredtiger_open(const char *home, WT_EVENT_HANDLER *event_handler,
 	__wt_scr_free(session, &i2);
 	__wt_scr_free(session, &i3);
 
+	if (free_merge_cfg)
+		__wt_free(session, merge_cfg);
```

Comment:
Yes, if the connection is not readonly, then `merge_cfg` is the final merged config that ends up in `conn->cfg` and that needs to stay for the lifetime of the connection.  For readonly, it is part 1 and the readonly config overrides are added in and we merge again.  In that case, `conn->cfg` is that new merged  config and we need to free `merge_cfg`.